raiden/transfer/channel.py

```diff
@@ -1778,6 +1918,76 @@ def handle_receive_withdraw_confirmation(
     return TransitionResult(channel_state, events)
 
 
+def handle_receive_withdraw_expired(
+    channel_state: NettingChannelState,
+    withdraw_expired: ReceiveWithdrawExpired,
+    block_number: BlockNumber,
+) -> TransitionResult:
+    events: List[Event] = list()
+
+    withdraw_states = filter(
+        lambda state: state.total_withdraw == withdraw_expired.total_withdraw,
+        channel_state.partner_state.withdraws,
+    )
+
+    if not withdraw_states:
+        invalid_withdraw_expired_msg = (
+            f"Withdraw expired of {withdraw_expired.total_withdraw} "
+            f"did not correspond to previous withdraw request"
+        )
+        return TransitionResult(
+            channel_state,
+            [
+                EventInvalidReceivedWithdrawExpired(
+                    attempted_withdraw=withdraw_expired.total_withdraw,
+                    reason=invalid_withdraw_expired_msg,
+                )
+            ],
+        )
+
+    # Take the first withdraw state found because
+    # withdraw expiry happens in order of which
+    # withdraws have been triggered.
```

Comment:
Not the first, but the only match, if any. This filter cannot match more then one entry because the withdraw is a strictly monotonic value.

Reply:
Removed "first"